### ISSUE TITLE
Incorrect syntax near the keyword 'option' on Batch Delete

### ISSUE THREAD

**Issue by nh43de** [zzzprojects/EntityFramework-Plus#161]
This error occurs on x64 build for versions 1.5.5, 1.5.4, 1.5.3, 1.5.2 .. as far back as 1.4.46 (did not test versions earlier).

Executed C#:
`apr.legacyTable_CustomCalcsSqls.Where(c => c.AccessDbName == accessDbName).Delete();`

Table SQL:
```
CREATE TABLE [legacyTable].[CustomCalcsSql] (
	[AccessDbName] VARCHAR(100) PRIMARY KEY,
	[CustomCalcsSql] VARCHAR(MAX)
)
```

Command Text: 
```
DECLARE @rowAffected INT
DECLARE @totalRowAffected INT

SET @totalRowAffected = 0

WHILE @rowAffected IS NULL
    OR @rowAffected > 0
    BEGIN
        DELETE TOP (4000)
        FROM    A
        FROM    [legacyTable].[CustomCalcsFieldsParsed] AS A
                INNER JOIN ( SELECT 
    1 AS [C1], 
    [Extent1].[AccessDbName] AS [AccessDbName], 
    [Extent1].[FieldName] AS [FieldName]
    FROM [legacyTable].[CustomCalcsFieldsParsed] AS [Extent1]
    WHERE [Extent1].[AccessDbName] = @p__linq__0 option (recompile)
                           ) AS B ON A.[AccessDbName] = B.[AccessDbName]
AND A.[FieldName] = B.[FieldName]

        SET @rowAffected = @@ROWCOUNT
        SET @totalRowAffected = @totalRowAffected + @rowAffected
    END

SELECT  @totalRowAffected
```

**Comment by JonathanMagnan** [comment]
Hello @nh43de ,

Thank you for reporting,

Are you using a third party library that makes the following text to appears `option (recompile)`? Currently, I don't think I have enough information to reproduce this issue since this line seems not coming directly from Entity Framework or our library.

Best Regards,

Jonathan

**Comment by nh43de** [comment]
Thanks Jonathan, 

Looks like one of our devs had thrown in a hint interceptor to the base repo class, causing this issue.

Closed #161

**Comment by JonathanMagnan** [comment]
Let me know if you want us to add an option to skip user interceptor.

Best Regards,

Jonathan
